cli/azd/pkg/project/framework_service_docker.go

```diff
@@ -25,7 +25,7 @@ type DockerProjectOptions struct {
 	Context   string           `json:"context"`
 	Platform  string           `json:"platform"`
 	Tag       ExpandableString `json:"tag"`
-	BuildArgs []string         `json:"buildArgs"`
+	BuildArgs []string         `json:"buildArgs" yaml:"buildArgs"`
```

Comment:
Why do we have the explicit `yaml` tag here but not on the other members?  This diff makes it seem like it was not working because this tag was missing, but the test makes it seem like at least some other properties were being loaded fine?  Should we add an assertion about `Tag` in the test?

Reply:
> Why do we have the explicit yaml tag here but not on the other members? 

I didn't find a good answer for this. It made me pull my hair for a few minutes, hahah.  
What I discovered by manual experimentation is that unmarshalling yaml would lower-case the field name by default. That's why it works for all the other fields (context, platform, etc). Because the field on `yaml` is one lower-cased word.

But, in case of `BuildArgs`, the only way it can work is by writing `buildargs` instead of `buildArgs` on the `yaml` file. But that goes against the yaml schema (so it is highlighted). But it works that way (that's currently the workaround for current version).

So, my conclusion is that we must use `yaml:"filed-name"` whenever the filed is not only lowercase.

Reply:
Ahh, yes.  https://pkg.go.dev/gopkg.in/yaml.v3#Unmarshal and https://pkg.go.dev/gopkg.in/yaml.v3#Marshal explain this a little better.  Thanks for the explanation and sorry for the loss of hair ;-).